modules/http/src/main/java/org/glassfish/grizzly/http/HttpCodecFilter.java

```diff
@@ -627,7 +631,19 @@ public final NextAction handleRead(final FilterChainContext ctx,
                 LOGGER.log(Level.FINE, "Error parsing HTTP header", e);
 
                 HttpProbeNotifier.notifyProbesError(this, connection, httpHeader, e);
-                onHttpHeaderError(httpHeader, ctx, e);
+
+                switch (parsingState.getHeaderParsingState().state)
+                {
+                    case 0:
```

Comment:
What are these parsing states? Isn't there a possibility to make this more declarative?

Reply:
The parsing state means in which state it failed during the HTTP parsing. This states are modified in the function decodeHttpPacketFromBytes. I know that if it failed in state 0 it's a 414 for example.

Reply:
Ok.